components/ai_chat/core/browser/engine/conversation_api_v2_client.cc

```diff
@@ -40,6 +68,67 @@ net::NetworkTrafficAnnotationTag GetNetworkTrafficAnnotationTag() {
 
 }  // namespace
 
+// static
+base::Value::List ConversationAPIV2Client::SerializeOAIMessages(
+    std::vector<OAIMessage> messages) {
+  static constexpr auto kTypeMap =
+      base::MakeFixedFlatMap<ExtendedContentBlockType, std::string_view>(
+          {{ExtendedContentBlockType::kText, "text"},
+           {ExtendedContentBlockType::kImage, "image_url"},
+           {ExtendedContentBlockType::kPageExcerpt, "brave-page-excerpt"},
+           {ExtendedContentBlockType::kChangeTone, "brave-request-change-tone"},
+           {ExtendedContentBlockType::kParaphrase, "brave-request-paraphrase"},
+           {ExtendedContentBlockType::kImprove,
+            "brave-request-improve-excerpt-language"},
+           {ExtendedContentBlockType::kShorten, "brave-request-shorten"},
+           {ExtendedContentBlockType::kExpand, "brave-request-expansion"}});
+
+  base::Value::List serialized_messages;
+  for (auto& message : messages) {
+    base::Value::Dict message_dict;
+
+    // Set role
+    message_dict.Set("role", std::move(message.role));
+
+    // Content blocks
+    base::Value::List content_list;
+    for (auto& extended_content_block : message.content) {
+      base::Value::Dict content_block_dict;
+
+      // Set type
+      auto type_it = kTypeMap.find(extended_content_block.type);
+      CHECK(type_it != kTypeMap.end());
+      content_block_dict.Set("type", type_it->second);
+
+      // Set content data based on variant type.
+      std::visit(
+          absl::Overload{
+              [&](TextContent& data) {
+                content_block_dict.Set("text", std::move(data.text));
+              },
+              [&](ImageContent& data) {
+                base::Value::Dict image_url;
+                image_url.Set("url", std::move(data.image_url.url));
+                if (data.image_url.detail) {
+                  image_url.Set("detail", std::move(*data.image_url.detail));
+                }
+                content_block_dict.Set("image_url", std::move(image_url));
+              },
+              [&](ChangeToneContent& data) {
+                content_block_dict.Set("text", "");
```

Comment:
we need a comment saying why we need this empty text holder.

Reply:
https://github.com/brave/brave-core/pull/32663/commits/fc4ddbaef89521628a54255888bb702ca9956baf